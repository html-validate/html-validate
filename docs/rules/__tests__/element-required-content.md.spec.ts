import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<html>
</html>`;
markup["correct"] = `<html>
    <head></head>
    <body></body>
</html>`;

describe("docs/rules/element-required-content.md", () => {
	it("inline validation: incorrect", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"element-required-content":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"element-required-content":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
