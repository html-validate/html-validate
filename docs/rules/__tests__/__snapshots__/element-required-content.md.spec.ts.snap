// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/element-required-content.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/element-required-content.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 2,
        "context": Object {
          "missing": "head",
          "node": "html",
        },
        "line": 1,
        "message": "<html> element must have <head> as content",
        "offset": 1,
        "ruleId": "element-required-content",
        "severity": 2,
        "size": 4,
      },
      Object {
        "column": 2,
        "context": Object {
          "missing": "body",
          "node": "html",
        },
        "line": 1,
        "message": "<html> element must have <body> as content",
        "offset": 1,
        "ruleId": "element-required-content",
        "severity": 2,
        "size": 4,
      },
    ],
    "source": "<html>
</html>",
    "warningCount": 0,
  },
]
`;
